src/log/zipkin.cpp

```diff
@@ -155,13 +166,38 @@ fc::variant create_zipkin_variant( zipkin_span::span_data&& span, const std::str
    return result;
 }
 
+void zipkin::post_request(zipkin_span::span_data&& span) {
+   boost::asio::post(my->work_strand, [this, span{std::move(span)}]() mutable {
+         my->log( std::move( span ) );
+   });
+}
+
 void zipkin::log( zipkin_span::span_data&& span ) {
-   if( my->consecutive_errors > my->max_consecutive_errors || my->stopped )
+   if( my->stopped ) {
       return;
+   }else if( fc::do_sighup.test_and_set()) { // member func .test() is not available until c++20.
+      fc::do_sighup.clear();
+      my->consecutive_errors = 0;
+   }else if( my->consecutive_errors > my->max_consecutive_errors ) {
+      fc::do_sighup.clear(); // reverse test_and_set()
+      return;
+   }
 
-   boost::asio::post(my->work_strand, [this, span{std::move(span)}]() mutable {
-      my->log( std::move( span ) );
-   });
+   fc::do_sighup.clear(); // reverse test_and_set()
```

Comment:
There is a race condition here on `do_sighup`. For example, if you SIGHUP  right before this `clear` call it will be ignored.

Reply:
Yes, there is a race condition. It would be perfect to use `atomic_flag` here if its member function test() were available in c++17.  Is it acceptable if I revert back to the previous solution - using two vars one for signal safe and the other for thread safe? The thread safe var` std::atomic<bool>` is lock free.